tests/test_api.py

```diff
@@ -467,6 +470,158 @@ def test_create_specification_auth(testclient):
+        if len(build_ids) < prev_builds:
+            if prev is not None:
+                build_delta = (now - prev).total_seconds()
+                print("build_delta", build_delta)
+                build_deltas.append(build_delta)
+            prev_builds = len(build_ids)
+            prev = now
+
+        # Gets the oldest build in the queue as it's the one that's most likely
+        # to be done
+        try:
+            build_id = build_ids.popleft()
+        except IndexError:
+            break
+
+        # Checks the status
+        response = testclient.get(f"api/v1/build/{build_id}", timeout=10)
+        response.raise_for_status()
+        r = schema.APIGetBuild.parse_obj(response.json())
+        assert r.status == schema.APIStatus.OK
+        assert r.data.specification.name == environment_name
+
+        # Exits immediately on failure
+        assert r.data.status != 'FAILED'
+
+        # If not done, adds the id back to the end of the queue
+        if r.data.status != 'COMPLETED':
+            build_ids.append(build_id)
+
+        # Adds a small delay to avoid making too many requests too fast. The
+        # build takes significantly longer than 1 second, so this shouldn't
+        # impact the measurements
+        time.sleep(1)
+
+    # If there are jobs in the queue, the loop didn't complete in the allocated
+    # time. So something went wrong, like a build getting stuck or parallel
+    # builds not working
+    assert len(build_ids) == 0
+
+    # Because this is an integration test, we cannot change the server
+    # c.CondaStoreWorker.concurrency value, which is set to 4 by default. But
+    # it's possible to devise a statistical test based on locally collected
+    # data, using 'build_deltas' above:
+    #
+    # concurrency = 4 with 2 CPUs, so equivalent to concurrency = 2:
+    # c4_2cpu = [
+    #     1.027987,
+    #     67.234371,
+    #     1.272288,
+    #     43.966526,
+    #     7.171627,
+    #     68.563222,
+    #     4.143675,
+    #     46.872263,
+    #     1.018258,
+    # ]
+    #
+    # concurrency = 1, same machine:
+    # c1 = [
+    #     19.394085,
+    #     33.70623,
+    #     22.815429,
+    #     62.555845,
+    #     68.438333,
+    #     29.794979,
+    #     63.370743,
+    #     32.118421,
+    #     29.88376,
+    # ]
+    #
+    # Here's another set of measurements from a different machine, which should
+    # have 4 CPUs, with concurrency = 4:
+    # ci = [
+    #     1.02644,
+    #     33.591736,
+    #     1.016269,
+    #     19.299738,
+    #     7.115025,
+    #     20.342442,
+    #     12.222805,
+    #     6.103751,
+    #     1.016237,
+    # ]
+    #
+    # These values will vary depending on workload and the number of CPUs. But
+    # the main observation here is this: if parallel builds are working,
+    # there will be a number of values that are relatively small compared to the
+    # time it takes to run a single build when not running concurrently.
```

Comment:
This is because these 1-5 second values are actually two builds running in parallel and completing almost at the same time after each other:

```
| build1 start -------------------------------- build1 done | <-> 1-5 second delta here
  | build2 start ----------------------------------- build2 done |
time ------------------------------------------------------------------------->
```